katdal/chunkstore.py

```diff
@@ -437,6 +441,58 @@ def put_dask_array(self, array_name, array, offset=()):
         out_chunks = tuple(len(c) * (1,) for c in array.chunks)
         return da.Array(dask_graph, out_name, out_chunks, np.object)
 
+    def list_chunk_ids(self, array_name):
+        """List all chunk ID strings associated with given array in chunk store.
+
+        Parameters
+        ----------
+        array_name : string
+            Identifier of array in chunk store
+
+        Returns
+        -------
+        chunk_ids : list of string
+            List of chunk identifier strings (e.g. '00012_01024_00000')
+        """
+        raise NotImplementedError
+
+    def has_array(self, array_name, chunks, offset=()):
```

Comment:
Should this maybe be a private function, with `has_dask_array` being the public, will-always-work frontend? Otherwise I can easily see someone accidentally using this version without being fully aware that they have to handle the not implemented case.

Reply:
I intend to use this in katdal one of these days, to verify the presence of all the chunks in the dataset. I was upset when I found out that a 10000-element `da.from_array`-based `dask.Array` can still take 7 seconds to express, so for some functionality I'd rather call `has_array`.

The main use of `has_dask_array` is when building up the flags dask array. Eventually the dataset selection will zoom in on a specific set of chunks to compute, and `da.map_blocks` (coupled with dask arrays for the flag components) simplifies this a lot.

The not implemented case is rados (pfft) and dict. The latter could still override `has_array` instead of `list_chunk_ids` to get efficiency.